sdks/wasm/Mono.WebAssembly.DebuggerProxy/MonoProxy.cs

```diff
@@ -219,6 +207,47 @@ protected override async Task<bool> AcceptCommand (MessageId id, string method,
 			return false;
 		}
 
+		async Task OnGetPossibleBreakpoints (MessageId id, string method, JObject args, CancellationToken token)
+		{
+			var startArg = args? ["start"] as JObject;
+			if (startArg == null) {
+				var err = Result.Err ("Protocol error: 'start' parameter missing.");
+				SendResponse (id, err, token);
+				return;
+			}
+
+			var start = SourceLocation.Parse (startArg);
+			if (start == null) {
+				// Not a dotnet:// location.
+				var resp = await SendCommand (id, method, args, token);
+				SendResponse (id, resp, token);
+				return;
+			}
+
+			//FIXME support variant where restrictToFunction=true and end is omitted
+			var end = SourceLocation.Parse (args? ["end"] as JObject);
+
+			var store = await RuntimeReady (id, token).ConfigureAwait (false);
+
+			List<SourceLocation> bps;
+			try {
+				bps = store.FindPossibleBreakpoints (start, end);
+				if (bps == null) {
+					var err = Result.Err ("Protocol error: cannot find any breakpoints.");
+					SendResponse (id, err, token);
+					return;
+				}
+			} catch (Exception ex) {
+				var err = Result.Exception (ex);
+				SendResponse (id, err, token);
```

Comment:
have you tested how vs and vcode deal with an error here?

Reply:
I'd have to check - I simply wanted to make it a bit more robust by not crashing the proxy with an unhandled exception on invalid input.